### PR TITLE
refactor orchestrate, terminate, restart functions

### PR DESCRIPTION
**Pull request by containscafeine**
Put redundant code under `def bootstrap_env` so `orchestrate`, `terminate`, `restart` functions could reuse it.

### CODE REVIEW

File: container/docker/engine.py [hunk 1]
```diff
@@ -780,3 +726,60 @@ def print_version_info(self):
         client = self.get_client()
         pprint.pprint(client.info())
         pprint.pprint(client.version())
+
+    def bootstrap_env(self, temp_dir, behavior, operation, compose_option,
+                      builder_img_id=None, context=None):
```
Comment (chouseknecht) [inline container/docker/engine.py]: Set context={}, rather than context=None. Then lines 746 - 747 can go away.

File: container/docker/engine.py [hunk 2]
```diff
@@ -780,3 +726,60 @@ def print_version_info(self):
...
+        """
+        Builds the environment for common operations for compose engine,
+        like orchestration, termination, etc.
```
Comment (chouseknecht) [inline container/docker/engine.py]: This sentence doesn't quite make sense. Should be something like: 

Build common Docker Compose elements required to execute orchestrate, terminate, restart, etc.

File: container/docker/engine.py (hunk 1 above)
Comment (containscafeine) [inline container/docker/engine.py]: I had done that, but then my IDE threw some warning about using mutable default arguments, and then I arrived at [this](http://docs.python-guide.org/en/latest/writing/gotchas/), so I decided to go with `context=None`.
WDYT?

File: container/docker/engine.py (hunk 2 above)
Comment (containscafeine) [inline container/docker/engine.py]: Fixed. Rebased.

**Pull request merged**